### PR TITLE
Adding attributes block to product block editor.

### LINKED ISSUE #37728: Product Block Editor: Create product attributes block
**Description**

This issue will cover the creation of the product attributes block as we had designed in the original implementation.
With a few small additional changes (these changes can be done within the component as we will be replacing the current implementation with the block editor soon):
- The add attribute modal contains a description below the title
- The `By default, new attributes are filterable and visible on the product page. You can change these settings for each attribute separately later.` notice shown at the top has been moved down.

**Items shown in the designs that aren't part of this issue**
- The entire right column in the add and edit modals (the preview)

Image

Image

**Designs**

AkVNGImLgSqCObTQ3idVn7-fi-1005_169648

**Acceptance criteria**

- [ ] Include a new **Attributes** section with the correct icon and description that contains the attributes field.
- [ ] **Add attribute** button should be present rendering a Modal to allow users to add attributes ( same as old design )
- [ ] Once attributes are added they can be edited, ~~re-ordered~~, removed (same as the old design).

### PR DESCRIPTION
**Pull request by joelclimbsthings**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
 
This moves all attributes components (there are many) to the product-editor package and adds a block container. Certain logic and styling changes are made to match the updated design.

Note that some of the changes also impact the old editor using the same component(s), in a slightly less than ideal but non-breaking way. This is considered fine since it will be deprecated very soon.

Also note that dragging does not work for the same reason as the images, and I'm a bit uncertain on how to handle thus far. 

Closes #37728 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout branch and enable `product-block-editor` feature flag with the WCA Test Helper plugin.
2. Navigate to Products -> Add new to view the new block editor.
3. At the bottom of the initial "General" tab you will find the attributes section.
4. Compare it to the design: `AkVNGImLgSqCObTQ3idVn7-fi-1761-446426&t=kYXzUnc4UoMZ91dd-4`
5. Try adding attributes, deleting them, and ensuring they persist after saving and refreshing the page.
6. Do not try dragging/reordering, as that functionality currently [does not work](https://github.com/woocommerce/woocommerce/issues/37728#issuecomment-1528128720).

### CODE REVIEW

File: packages/js/product-editor/package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
 		"@woocommerce/components": "workspace:*",
 		"@woocommerce/currency": "workspace:*",
 		"@woocommerce/customer-effort-score": "workspace:*",
-		"@woocommerce/data": "workspace:^4.1.0",
+		"@woocommerce/data": "workspace:*",
```
Comment (joshuatf) [inline packages/js/product-editor/package.json:38]: 🎉

File: packages/js/product-editor/src/components/attribute-empty-state/attribute-empty-state.tsx [hunk 2]
```diff
@@ -19,7 +18,7 @@ type AttributeEmptyStateProps = {
 };
 
 export const AttributeEmptyState: React.FC< AttributeEmptyStateProps > = ( {
-	image = AttributeEmptyStateLogo,
+	image,
```
Comment (joshuatf) [inline packages/js/product-editor/src/components/attribute-empty-state/attribute-empty-state.tsx]: Any reason for removing the default image here?  Looks like we're also no longer using this component.

File: packages/js/product-editor/src/components/attribute-empty-state/attribute-empty-state.tsx [hunk 3]
```diff
@@ -19,7 +18,7 @@ type AttributeEmptyStateProps = {
 };
 
 export const AttributeEmptyState: React.FC< AttributeEmptyStateProps > = ( {
```
Comment (joshuatf) [inline packages/js/product-editor/src/components/attribute-empty-state/attribute-empty-state.tsx]: Any reason we need to keep this component if it's no longer in use?

**Review by joshuatf — DISMISSED** [review]
Looks great!  Left one question about an unused component, but pre-approving.

Noting that the drag and drop functionality does not work here, but I think we should get this merged and continue to iterate on this experience once we've made some decisions around how we want to handle that.

File: packages/js/product-editor/src/components/attribute-empty-state/attribute-empty-state.tsx (hunk 2 above)
Comment (joelclimbsthings) [reply packages/js/product-editor/src/components/attribute-empty-state/attribute-empty-state.tsx]: Nope! Not a good one. Just removed that component in https://github.com/woocommerce/woocommerce/pull/38051/commits/9048f92843b922d9741477152645f4dc07118753.

File: packages/js/product-editor/src/components/attribute-empty-state/attribute-empty-state.tsx (hunk 3 above)
Comment (joelclimbsthings) [reply packages/js/product-editor/src/components/attribute-empty-state/attribute-empty-state.tsx]: Nope, just removed in https://github.com/woocommerce/woocommerce/pull/38051/commits/9048f92843b922d9741477152645f4dc07118753 👍🏻

**Comment by joelclimbsthings** [comment]
Thanks @joshuatf , just removed that component as suggested. 👍🏻

**Review by joshuatf — APPROVED** [review]
Thanks for removing that extra component.  LGTM!

**Pull request merged**